database/raft/state/state.go

```diff
@@ -0,0 +1,227 @@
+package state
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/golang/protobuf/proto"
+
+	"chain/database/raft/internal/statepb"
+	"chain/errors"
+	"chain/log"
+)
+
+const nextNodeID = "raft/nextNodeID"
+
+var ErrAlreadyApplied = errors.New("entry already applied")
+
+// State is a general-purpose data store designed to accumulate
+// and apply replicated updates from a raft log.
+// The zero value is an empty State ready to use.
```

Comment:
I don't think this is true, because the `map`s will be uninitialized (by either `make` or with a literal).